prow/jira/jira.go

```diff
@@ -214,11 +263,49 @@ func (jc *client) JiraURL() string {
 	return jc.url
 }
 
+// Used determines whether the client has been used
+func (jc *client) Used() bool {
+	return jc.clientUsed.Used()
+}
+
 type bearerAuthRoundtripper struct {
 	generator BearerAuthGenerator
 	upstream  http.RoundTripper
 }
 
+// WithFields clones the client, keeping the underlying delegate the same but adding
+// fields to the logging context
+func (jc *client) WithFields(fields logrus.Fields) Client {
+	return &client{
+		clientUsed: jc.clientUsed,
+		upstream:   jc.upstream,
+		logger:     jc.logger.WithFields(fields),
+		delegate:   jc.delegate,
+	}
+}
+
+// ForPlugin clones the client, keeping the underlying delegate the same but adding
+// a plugin identifier and log field
+func (jc *client) ForPlugin(plugin string) Client {
```

Comment:
There is a bug somewhere in that function or something is not initialized for integration tests as they are failing on nil. I'll try to investigate later on.

Reply:
Discovered that jiraClient can be nil in Agent. Added if that should solve the problem.